### PR TITLE
Code refactoring for KNN

### PR DESCRIPTION
**Pull request by hpandeycodeit**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/160/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/161/

File: src/ports/postgres/modules/knn/knn.py_in [hunk 1]
```diff
@@ -127,5 +124,102 @@ def knn_validate_src(schema_madlib, point_source, point_column_name, label_colum
...
+    if operation == 'c':
+        returnstring = 'The classification results have been written to output table '+ output_table;
+    else:
+        returnstring = 'The regression results have been written to output table '+ output_table;
+
+    plpy.execute("""DROP TABLE pg_temp.{madlib_knn_interm}""".format(**locals()));    
+
+    return returnstring;
+
+
```
Comment (iyerr3) [inline src/ports/postgres/modules/knn/knn.py_in]: In general, file contains multiple style-guide and PEP8 issues. Please refer to [page in wiki for appropriate style](https://cwiki.apache.org/confluence/display/MADLIB/Python+Style+Guide).

File: src/ports/postgres/modules/knn/knn.py_in [hunk 2]
```diff
@@ -127,5 +124,102 @@ def knn_validate_src(schema_madlib, point_source, point_column_name, label_colum
...
+        KNN function to find the K Nearest neighbours
+        Args:
+            @param schema_madlib       Name of the Madlib Schema
+            @param point_source        Training data table 
+            @param point_column_name   Name of the column with training data points.
+            @param label_column_name   Name of the column with labels/values of training data points.
+            @param test_source         Name of the table containing the test data points.
+            @param test_column_name    Name of the column with testing data points.
+            @param id_column_name      Name of the column having ids of data points in test data table.
+            @param output_table        Name of the table to store final results.
```
Comment (hpandeycodeit) [inline src/ports/postgres/modules/knn/knn.py_in]: added operation details.

File: src/ports/postgres/modules/knn/knn.py_in [hunk 3]
```diff
@@ -127,5 +124,102 @@ def knn_validate_src(schema_madlib, point_source, point_column_name, label_colum
...
+            @param k                   default: 1. Number of nearest neighbors to consider
+
+
+        Returns: 
+            VARCHAR                     Name of the output table.             
+    """                                
+
+  
+    oldClientMinMessages = plpy.execute("SELECT setting FROM pg_settings WHERE name = 'client_min_messages'")[0]['setting'];
```
Comment (hpandeycodeit) [inline src/ports/postgres/modules/knn/knn.py_in]: This is also done.

File: src/ports/postgres/modules/knn/knn.py_in [hunk 4]
```diff
@@ -127,5 +124,102 @@ def knn_validate_src(schema_madlib, point_source, point_column_name, label_colum
...
+
+    plpy.execute("SELECT {schema_madlib}.create_schema_pg_temp()".format(schema_madlib = schema_madlib));
+ 
+    x_temp_table = unique_string(desp='x_temp_table') 
+    y_temp_table = unique_string(desp='y_temp_table') 
+    label_column_name_unique = unique_string(desp='label_column_name_unique')  
+    test_id = unique_string(desp='test_id')  
+
+    convert_boolean_to_int = '';
+    if operation == 'c':
```
Comment (hpandeycodeit) [inline src/ports/postgres/modules/knn/knn.py_in]: yes, I used created a boolean flag for this comparison.

File: src/ports/postgres/modules/knn/knn.py_in (hunk 1 above)
Comment (hpandeycodeit) [inline src/ports/postgres/modules/knn/knn.py_in]: I did some changes as per the style guide. Will fix it all in a day or two

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/179/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/184/

**Comment by orhankislal** [comment]
jenkins ok to test

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/193/

**Pull request closed without merging**